main/client/src/mill/main/client/MillClientMain.java

```diff
@@ -122,10 +134,15 @@ public void run() {
                             args,
                             env
                     );
+
+                    // Should we give the server process some extra time to flush the output files?
+                    // currently, we don't
```

Comment:
If we call `System.out.flush()` and `System.err.flush()` on the server before closing the `serverSocket`, I think that would be enough to ensure that both output streams are flushed to disk